### PR TITLE
New PR for 

### PR DESCRIPTION
**Pull request by muschellij2**
This replaced #411 PR.  Removed extraneous .rproj edits and such.

Once checks are done - this can be merged.

Fix reddit demo where user_agent was being overloaded. Now can pass user_agent using config_init to init_oauth2.0 and Token2.0$new() (@muschellij2 @hadley #363 and #413).

Changed example in write_stream from https://jeroenooms.github.io/data/diamonds.json to https://github.com/jeroen/data/raw/gh-pages/diamonds.json as this had moved (@muschellij2).

Also, fixed an example on write_stream: https://github.com/r-lib/httr/blob/1078b1d35d0c5d0a61d53461cc6c37f9fb49551d/R/write-function.R#L61 as that link was dead and causing errors in the "examples" part of R CMD check

### CODE REVIEW

**Comment by hadley** [comment]
Can you please move the `write_stream` stuff to a separate PR?

**Comment by muschellij2** [comment]
OK - this is split.  After merged - I will fetch, then send new PR.

**Comment by muschellij2** [comment]
All checks passed - ready to merge.

File: R/oauth-init.R [hunk 1]
```diff
@@ -62,12 +62,14 @@ init_oauth1.0 <- function(endpoint, app, permission = NULL,
 #'     retrieve the token. Some authorization servers require this.
 #'     If \code{FALSE}, the default, retrieve the token by including the
 #'     app key and secret in the request body.
+#' @param config_init Additional configuration settings sent to
+#' \code{\link{POST}}, e.g. \code{\link{user_agent}}.
```
Comment (hadley) [inline R/oauth-init.R]: Needs to be indented - http://style.tidyverse.org/code-documentation.html#indention

File: R/oauth-init.R [hunk 2]
```diff
@@ -109,10 +111,11 @@ init_oauth2.0 <- function(endpoint, app, scope = NULL, user_params = NULL,
 
   if (isTRUE(use_basic_auth)) {
     req <- POST(endpoint$access, encode = "form", body = req_params,
-      authenticate(app$key, app$secret, type = "basic"))
+      authenticate(app$key, app$secret, type = "basic"), config = config_init)
```
Comment (hadley) [inline R/oauth-init.R]: Can you please convert all the calls you touched to use tidyverse style - http://style.tidyverse.org/syntax.html#long-lines

File: demo/oauth2-reddit.R [hunk 3]
```diff
@@ -15,3 +15,11 @@ token <- oauth2.0_token(reddit, app,
   scope = c("read", "modposts"),
   use_basic_auth = TRUE
 )
+
+# 3b. If get 429 too many requests, the default user_agent is overloaded.
+# If you have an application on Reddit then you can pass that using:
+token <- oauth2.0_token(reddit, app,
```
Comment (hadley) [inline demo/oauth2-reddit.R]: Please only use two spaces to indent - http://style.tidyverse.org/syntax.html#long-lines

**Comment by muschellij2** [comment]
Quite the hassle for such a small PR!  It's stylized now.

File: R/oauth-init.R [hunk 4]
```diff
@@ -109,10 +111,12 @@ init_oauth2.0 <- function(endpoint, app, scope = NULL, user_params = NULL,
 
   if (isTRUE(use_basic_auth)) {
     req <- POST(endpoint$access, encode = "form", body = req_params,
-      authenticate(app$key, app$secret, type = "basic"))
+      authenticate(app$key, app$secret, type = "basic"),
```
Comment (hadley) [inline R/oauth-init.R]: This still isn't indented correctly. It should look like

```R
req <- POST(endpoint$access, 
   encode = "form", 
   body = req_params,
   authenticate(app$key, app$secret, type = "basic"),
   config = config_init
)
```

Comment (muschellij2) [inline R/oauth-init.R]: I've made the edits and pushed.  Why is RStudio `Reindent-Lines` not `tidyverse`-style-compatible?

**Comment by hadley** [comment]
Thanks!

**Pull request merged**